testing/file_test/file_test_base.h

```diff
@@ -77,7 +77,7 @@ class FileTestBase : public testing::Test {
   // The return value should be an error if there was an abnormal error, and
   // RunResult otherwise.
   virtual auto Run(const llvm::SmallVector<llvm::StringRef>& test_args,
-                   llvm::vfs::InMemoryFileSystem& fs,
+                   llvm::IntrusiveRefCntPtr<llvm::vfs::InMemoryFileSystem> fs,
```

Comment:
Can you elaborate why you're generally passing by value instead of by reference? That is, I understand a IntrusiveRefCntPtr is needed for ClangRunner. However, do we benefit from doing reference counting at each call boundary?

Reply:
I did not realize there was a performance concern here, and passing by value provides a simpler contract in general. I will make more of these references.

Reply:
Yes, these have complex construction/destruction so don't fall into the "trivial struct" case of copying.